x-pack/legacy/plugins/maps/public/layers/sources/source_registry.ts

```diff
@@ -0,0 +1,35 @@
+/*
```

Comment:
There is forward-thinking reasoning behind introducing the source-registry, that this could serve as an extension for 3rd party plugins to register custom sources.

But this isn't strictly necessary to achieve what we need in the short term:
- in this PR, to enable the split between clusters/heatmaps
- going forward to the more generic "curated layers" concept

I'm thinking we could drop this introduction from this PR as it just introduces additional abstractions.

Reply:
source_registry is needed because layer_wizard_registry does not correlate one-to-one with sources. For example, there are 2 wizards that return geo_grid source. 

all_sources is clunky and will not support or future needs.